scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -279,3 +280,57 @@ def get_generated_keys(self) -> List[str]:
         for operand in self._operands:
             generated_keys += operand.get_generated_keys()
         return generated_keys
+
+
+class AttackPointIteratorCartesianProduct(AttackPointIterator):
+    """
+    Attack point iterator cartesian product class. This class takes any amount
+    of operands and combines them just like a cartesian product would.
+    """
+
+    def __init__(self, operation: str, operands: List[Dict[str, Any]]) -> None:
+        """Initialize the cartesian product iterator.
+          
+          Args:
+            operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                
+            operands (List[Dict[str, Any]]): The operands are any number of
+                iterator configs that will be combined."""
+        assert "cartesian_product" == operation
+        self._operands = list(
+            build_attack_points_iterator(operand) for operand in operands)
+
+        operand_lengths = [operand._len for operand in self._operands]
+        if not self._operands or any(
+                length <= 0
+                for length in operand_lengths) or len(operand_lengths) != 2:
```

Comment:
A cartesian product of a single operand should iterate that operand.

Reply:
If there are three or more operands you can do an easy trick:

```
self._operands = [build_attack_points_iterator(operands[0]), build_attack_points_iterator({"operation": "cartesian_product", "operands": operands[1:]})]
```

This requires some refactoring here and there. Also I believe that unit-tests should have three or four operands and they currently don't have those.

Reply:
what is the difference between this to the original one?

Reply:
Oh ok I see but then you would have to check if length is longer then 1 right?